src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicAnalysisDataReader.cs

```diff
@@ -0,0 +1,367 @@
+            int resourceSize = *(int*)resourceStart;
+            if (resourceSize > resourcesDir.Size - sizeof(int))
+            {
+                throw new BadImageFormatException();
+            }
+
+            return new DynamicAnalysisDataReader(resourceStart + sizeof(int), resourceSize);
+        }
+
+        private static int ReadReference(ref BlobReader reader, bool smallRefSize)
+        {
+            return smallRefSize ? reader.ReadUInt16() : reader.ReadInt32();
+        }
+
+        public ImmutableArray<LinePositionSpan> GetSpans(BlobHandle handle)
+        {
+            var builder = ArrayBuilder<LinePositionSpan>.GetInstance();
+
+            var reader = GetBlobReader(handle);
+
+            // header:
+            int documentRowId = ReadDocumentRowId(ref reader);
+            int previousStartLine = -1;
+            ushort previousStartColumn = 0;
+
+            // records:
+            while (reader.RemainingBytes > 0)
+            {
+                int deltaLines, deltaColumns;
+                ReadDeltaLinesAndColumns(ref reader, out deltaLines, out deltaColumns);
+
+                // document:
+                if (deltaLines == 0 && deltaColumns == 0)
+                {
+                    documentRowId = ReadDocumentRowId(ref reader);
+                    continue;
+                }
+
+                int startLine;
+                ushort startColumn;
+
+                // delta Start Line & Column:
+                if (previousStartLine < 0)
+                {
+                    Debug.Assert(previousStartColumn == 0);
+
+                    startLine = ReadLine(ref reader);
+                    startColumn = ReadColumn(ref reader);
+                }
+                else
+                {
+                    startLine = AddLines(previousStartLine, reader.ReadCompressedSignedInteger());
+                    startColumn = AddColumns(previousStartColumn, reader.ReadCompressedSignedInteger());
+                }
+
+                previousStartLine = startLine;
+                previousStartColumn = startColumn;
+
+                int endLine = AddLines(startLine, deltaLines);
+                int endColumn = AddColumns(startColumn, deltaColumns);
+                builder.Add(new LinePositionSpan(new LinePosition(startLine, startColumn), new LinePosition(endLine, endColumn)));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        //TODO: some of the helpers below should be provided by System.Reflection.Metadata
+
+        private unsafe struct Blob
+        {
+            public readonly byte* Pointer;
+            public readonly int Length;
+
+            public Blob(byte* pointer, int length)
+            {
+                Pointer = pointer;
+                Length = length;
+            }
+        }
+
+        public Guid GetGuid(GuidHandle handle)
+        {
+            if (handle.IsNil)
+            {
+                return default(Guid);
```

Comment:
Is this an optimization to avoid encoding the default GUID?  Is it important to be able to distinguish between no-GUID and default-GUID?